src/EditorFeatures/Core/Shared/Tagging/EventSources/TaggerEventSources.ParseOptionChangedEventSource.cs

```diff
@@ -38,11 +38,15 @@ private void OnWorkspaceChanged(object sender, WorkspaceChangeEventArgs e)
                     if (!object.Equals(oldProject.ParseOptions, newProject.ParseOptions))
                     {
                         var workspace = e.NewSolution.Workspace;
-                        var documentIds = workspace.GetRelatedDocumentIds(SubjectBuffer.AsTextContainer());
-
-                        if (documentIds.Any(d => d.ProjectId == e.ProjectId))
+                        var documentId = workspace.GetDocumentIdInCurrentContext(SubjectBuffer.AsTextContainer());
+                        if (documentId != null)
                         {
-                            this.RaiseChanged();
+                            var relatedDocumentIds = e.NewSolution.GetRelatedDocumentIds(documentId);
+
+                            if (relatedDocumentIds.Any(d => d.ProjectId == e.ProjectId))
+                            {
+                                RaiseChanged();
```

Comment:
It's probably fine but this does meant now that RaiseChanged() could get called if a later workspace change has entirely removed the document. Previously the implicit CurrentSolution might have "fixed" that (or a least made the race window smaller.)

Reply:
This is still probably saner but if we see anything strange this might be a good place to look.

Reply:
Yeah. Having two race conditions seems worse than having just one :).